### ISSUE TITLE
Add extended media previews for messages

### ISSUE THREAD

**Issue by klinker24** [klinker-apps/messenger-issues#112]
**Request:**

Some messages contain content that could use more of an extended preview. For example:
- YouTube videos
- Web links
- Tweets
- Any other suggestions?

**Implementation notes:**

Create a new table for `extended_media`. Could just have a foreign key to the `message` table, then a `title`, `data`, and `media_type` column.

It would be trivial to do a join on the `message` table when querying the database then.

**Comment by klinker24** [comment]
I started doing Youtube today, and finished them. Integration is pretty easy, make a `MediaParser` class that can grab metadata from the media, add a mime type for that media, handle that mime type in the adapter.

The implementation is greatly simplified (and improved!) from the initial comment. It works by just inserting a new message into the database with the `type` column as `TYPE_MEDIA` and `mime_type` describing how it should be displayed in the adapter. We can insert anything into the `data` column. For example, with youtube, i just put the thumbnail link there, with web articles, it should be a json string with metadata like link, title, snippet, image link, etc.

On the web side, I was able to handle the `media/youtube` mime type just from the `messages.js` file, Others should be the same.

- [ ] TODO: Make clicking a youtube video from the web open up the video

This feature is feature toggled out at this time, so it will not be going to users yet.
